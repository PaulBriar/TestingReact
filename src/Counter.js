import React from 'react';

class Counter extends React.Component {
	state = { count: 0 };

	makeIncrementer = amount => () =>
		this.setState(prevState => ({
				count: prevState.count + amount,
		}));

  render() {
    return (
			<div>
				<p>Current Count: {this.state.count}</p>
				<button className="inc" onClick={this.makeIncrementer(1)}>Increment</button>
				<button className="dec" onClick={this.makeIncrementer(-1)}>Decrement</button>
			</div>
    )
  }
}

export default Counter;