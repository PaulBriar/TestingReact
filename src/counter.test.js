import React from 'react';
import Counter from './Counter';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Counter component', () => {
    it('Matches the snapshot', () => {
        const tree = renderer.create(<Counter />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Starts at 0', () => {
        const wrapper = shallow(<Counter />);
        const text = wrapper.find('p').text();
        expect(text).toEqual('Current Count: 0');
    });

    it('Can increment count when button pressed', () => {
        const wrapper = shallow(<Counter />);
        const incButton = wrapper.find('button.inc');
        incButton.simulate('click');
        const text = wrapper.find('p').text();
        expect(text).toEqual('Current Count: 1');
    });

    it('Can decrement count when button pressed', () => {
        const wrapper = shallow(<Counter />);
        const decButton = wrapper.find('button.dec');
        decButton.simulate('click');
        const text = wrapper.find('p').text();
        expect(text).toEqual('Current Count: -1');
    });
});