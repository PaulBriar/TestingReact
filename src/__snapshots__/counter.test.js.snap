// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Matches the snapshot 1`] = `
<div>
  <p>
    Current Count: 
    0
  </p>
  <button
    className="inc"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    className="dec"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
